ramda/ramda.d.ts

```diff
@@ -0,0 +1,1807 @@
+// Definitions by: Erwin Poeze <https://github.com/donnut>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+declare var R: R.Static;
+
+declare namespace R {
+    type Ord = number | string | boolean;
+
+    interface ListIterator<T, TResult> {
+        (value: T, index: number, list: T[]): TResult;
+    }
+
+    interface Functor<T> {
+        map<T>(a: any): T;
+    }
+
+    interface ObjectIterator<T, TResult> {
+        (element: T, key: string, obj: Dictionary<T>): Dictionary<TResult>;
+    }
+
+    interface KeyValuePair<K, V> extends Array<K | V> { 0 : K; 1 : V; }
+
+    interface ArrayLike {
+        nodeType: number;
+    }
+
+    interface Arity0Fn {
+        (): any
+    }
+
+    interface Arity1Fn {
+        (a: any): any
+    }
+
+    interface Arity2Fn {
+        (a: any, b: any): any
+    }
+
+    interface ObjFunc {
+        [index:string]: Function;
+    }
+
+    interface ObjFunc2 {
+        [index:string]: (x: any, y: any) => boolean;
+    }
+
+    interface Pred {
+        (...a: any[]): boolean;
+    }
+
+    interface ObjPred {
+        (value: any, key: string): boolean;
+    }
+
+    interface Dictionary<T> {
+        [index: string]: T;
+    }
+
+    interface CharList extends String {
+        push(x: string): void;
+    }
+
+    interface Nested<U> {
+        [index: string]: Nested<U>|{<U>(value: any): U};
+    }
+
+    interface Lens {
+        <T,U>(obj: T): U;
+        set<T,U>(str: string, obj: T): U;
+    }
+
+    // @see https://gist.github.com/donnut/fd56232da58d25ceecf1, comment by @albrow
+    interface CurriedFunction2<T1, T2, R> {
+        (t1: T1): (t2: T2) => R;
+        (t1: T1, t2: T2): R;
+    }
+
+    interface CurriedFunction3<T1, T2, T3, R> {
+        (t1: T1): CurriedFunction2<T2, T3, R>;
+        (t1: T1, t2: T2): (t3: T3) => R;
+        (t1: T1, t2: T2, t3: T3): R;
+    }
+
+    interface CurriedFunction4<T1, T2, T3, T4, R> {
+        (t1: T1): CurriedFunction3<T2, T3, T4, R>;
+        (t1: T1, t2: T2): CurriedFunction2<T3, T4, R>;
+        (t1: T1, t2: T2, t3: T3): (t4: T4) => R;
+        (t1: T1, t2: T2, t3: T3, t4: T4): R;
+    }
+
+    interface CurriedFunction5<T1, T2, T3, T4, T5, R> {
+        (t1: T1): CurriedFunction4<T2, T3, T4, T5, R>;
+        (t1: T1, t2: T2): CurriedFunction3<T3, T4, T5, R>;
+        (t1: T1, t2: T2, t3: T3): CurriedFunction2<T4, T5, R>;
+        (t1: T1, t2: T2, t3: T3, t4: T4): (t5: T5) => R;
+        (t1: T1, t2: T2, t3: T3, t4: T4, t5: T5): R;
+    }
+
+    interface CurriedFunction6<T1, T2, T3, T4, T5, T6, R> {
+        (t1: T1): CurriedFunction5<T2, T3, T4, T5, T6, R>;
+        (t1: T1, t2: T2): CurriedFunction4<T3, T4, T5, T6, R>;
+        (t1: T1, t2: T2, t3: T3): CurriedFunction3<T4, T5, T6, R>;
+        (t1: T1, t2: T2, t3: T3, t4: T4): CurriedFunction2<T5, T6, R>;
+        (t1: T1, t2: T2, t3: T3, t4: T4, t5: T5): (t6: T6) => R;
+        (t1: T1, t2: T2, t3: T3, t4: T4, t5: T5, t6: T6): R;
+    }
+
+    interface Reduced<T> {}
```

Comment:
This is identical to`{}` you need to use the `T` somewhere in a property otherwise the type checker ignores it. (See [structural typing](https://github.com/Microsoft/TypeScript/wiki/FAQ#why-is-astring-assignable-to-anumber-for-interface-at--)).